src/data/finsupp/lex.lean

```diff
@@ -0,0 +1,96 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.pi.lex
+import data.finsupp.order
+
+/-!
+# Lexicographic order on finitely supported functions
+
+This file defines the lexicographic order on `finsupp`.
+-/
+
+namespace finsupp
+
+variables {α N : Type*} [has_zero N]
+
+@[simp] lemma to_lex_apply (x : α →₀ N) (a : α) : to_lex (x a) = x a := rfl
+@[simp] lemma of_lex_apply (x : α →₀ N) (a : α) : of_lex (x a) = x a := rfl
```

Comment:
This is the wrong statement!
```suggestion
@[simp] lemma to_lex_apply (x : α →₀ N) (a : α) : to_lex x a = x a := rfl
@[simp] lemma of_lex_apply (x : lex (α →₀ N)) (a : α) : of_lex x a = x a := rfl
```

Reply:
I am confused by these: the current version does not typecheck and I can't seem to find a way that it would...